org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ClasspathManagement.scala

```diff
@@ -466,32 +476,54 @@ trait ClasspathManagement extends HasLogger { self: ScalaProject =>
 
         // create the classpath problem markers
         errors foreach {
-          case (severity, message, markerId) => (new cpMarkerFactory(markerId)).create(underlying, severity, message)
+          case ClasspathErrorMarker(severity, message, markerId) => ClasspathMarkerFactory(markerId).create(underlying, severity, message)
         }
       }
       Status.OK_STATUS
     }
   }
 
-  private def validateBinaryVersionsOnClasspath(): Seq[(Int, String, String)] = {
-    val entries = scalaClasspath.userCp
-    val errors = mutable.ListBuffer[(Int, String, String)]()
-    val badEntries = mutable.ListBuffer[(IPath, ScalaVersion)]()
-
-    for (entry <- entries if entry ne null) {
-      entry.lastSegment() match {
-        case VersionInFile(version) =>
-          if (!ScalaPlugin().isCompatibleVersion(version, this)) {
-            badEntries += ((entry,version))
-            val msg = s"${entry.lastSegment()} of ${this.underlying.getName()} build path is cross-compiled with an incompatible version of Scala (${version.unparse}). In case this report is mistaken, this check can be disabled in the compiler preference page."
-            errors += ((IMarker.SEVERITY_ERROR, msg, SdtConstants.ScalaVersionProblemMarkerId))
-          }
-        case _ =>
+  private def validateBinaryVersionsOnClasspath(): Seq[ClasspathErrorMarker] = {
+    scalaClasspath.userCp.foldLeft(List[ClasspathErrorMarker]()) {
+      case (errors, entry) ⇒ entry.lastSegment() match {
```

Comment:
This refactoring misses the null-check. It was necessary, according to 9b44832551d465c7f32916a113f9d5f5b5086ed3